### DISCUSSION TITLE
In controlnet, why is there a  `sample + controlnet_cond` preprocess step. I don't see this mention in the paper

### DISCUSSION THREAD

**Discussion by tinder-trideeprath** [huggingface/diffusers#7397] [Q&A]
This preprocess step is not clear, what is the requirement for this ? 

https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/controlnet.py#L798

```
        sample = self.conv_in(sample)

        controlnet_cond = self.controlnet_cond_embedding(controlnet_cond)
        sample = sample + controlnet_cond

```

**Comment by sayakpaul** [comment]
You can refer to the original implementation of ControlNet to confirm that.
